src/cpp/rtps/messages/MessageReceiver.cpp

```diff
@@ -406,6 +419,47 @@ bool MessageReceiver::readSubmessageHeader(CDRMessage_t* msg, SubmessageHeader_t
     return true;
 }
 
+bool MessageReceiver::willAReaderAcceptMsgDirectedTo(const EntityId_t & readerID)
+{
+    if(AssociatedReaders.empty()) {
+        logWarning(RTPS_MSG_IN,IDSTRING"Data received when NO readers are listening");
+        return false;
+    }
+
+    if (readerID != c_EntityId_Unknown) {
+        const auto readers = AssociatedReaders.find(readerID);
+        if (readers != AssociatedReaders.end() && !readers->second.empty()) {
```

Comment:
```suggestion
        if (readers != AssociatedReaders.end()) {
```
As we are now removing the entry when the vector becomes empty, the last check is not necessary